### PR TITLE
chore: rename atom -> electron

### PR DESCRIPTION
**Pull request by nornagon**
#### Description of Change
This renames all atom_* files to electron_*, and all "AtomFoo" classes to "ElectronFoo".

#### Checklist

- [x] PR description included and stakeholders cc'd
- [ ] `npm test` passes
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: none

### CODE REVIEW

File: chromium_src/chrome/browser/process_singleton_win.cc [hunk 1]
```diff
@@ -250,7 +250,8 @@ void ProcessSingleton::OnBrowserReady() {}
 // isn't one, create a message window with its title set to the profile
 // directory path.
 bool ProcessSingleton::Create() {
-  static const wchar_t kMutexName[] = L"Local\\AtomProcessSingletonStartup!";
+  static const wchar_t kMutexName[] =
+      L"Local\\ElectronProcessSingletonStartup!";
```
Comment (nornagon) [inline chromium_src/chrome/browser/process_singleton_win.cc]: I'm not sure what the implications of changing this are... cc @MarshallOfSound maybe?

Comment (nornagon) [reply chromium_src/chrome/browser/process_singleton_win.cc]: reverted this for now so as not to block.

Comment (MarshallOfSound) [reply chromium_src/chrome/browser/process_singleton_win.cc]: @nornagon Can we add a comment here to not change this without prior planning, if we change this we'll need to support both names for a while as we want the singleton to work across app updates

Comment (nornagon) [reply chromium_src/chrome/browser/process_singleton_win.cc]: I just reverted the change, we can deal with it separately.

**Comment by zcbenz** [comment]
Linux build fails because of `::Atom` type getting renamed:

```
../../electron/shell/browser/ui/x/x_window_utils.h:14:3: error: no type named 'Electron' in the global namespace
::Electron GetAtom(const char* name);
~~^
../../electron/shell/browser/ui/x/x_window_utils.h:18:55: error: no type named 'Electron' in the global namespace
void SetWMSpecState(::Window xwindow, bool enabled, ::Electron state);
                                                    ~~^
../../electron/shell/browser/native_window_views.cc:222:17: error: no member named 'Electron' in the global namespace
  std::vector<::Electron> state_atom_list;
              ~~^
../../electron/shell/browser/native_window_views.cc:222:27: error: C++ requires a type specifier for all declarations
  std::vector<::Electron> state_atom_list;
                          ^
```

**Comment by zcbenz** [comment]
The Linux CI is crashing at printer related code, but it does not seem related to this PR as the master branch also has this failure:
https://circleci.com/gh/electron/electron/454187#tests/containers/2

```
[454:0131/234108.821251:FATAL:web_contents_user_data.h(47)] Check failed: contents. 
#0 0x00005bbb673f base::debug::CollectStackTrace()
#1 0x00005bae2ee2 base::debug::StackTrace::StackTrace()
#2 0x00005baf90a1 logging::LogMessage::~LogMessage()
#3 0x000058977ec5 electron::api::WebContents::OnGet*******Printer()
#4 0x0000589896d2 base::internal::FunctorTraits<>::Invoke<>()
#5 0x000058989595 base::internal::Invoker<>::RunOnce()
#6 0x000058989a5b base::internal::ReplyAdapter<>()
#7 0x00005890f916 base::internal::Invoker<>::RunOnce()
#8 0x00005bb7df10 base::(anonymous namespace)::PostTaskAndReplyRelay::RunReply()
#9 0x00005bb7dfcd base::internal::Invoker<>::RunOnce()
#10 0x00005bb3d062 base::TaskAnnotator::RunTask()
```
/cc @codebytere

**Comment by zcbenz** [comment]
This PR requires review from @electron/wg-upgrades.

**Comment by deepak1556** [comment]
Linux crash tracked in https://github.com/electron/electron/pull/21956/files#r374235151

**Pull request merged**